lib/http/2/connection.rb

```diff
@@ -450,7 +450,14 @@ def connection_management(frame)
           @state = :closed
           @closed_since = Time.now
           emit(:goaway, frame[:last_stream], frame[:error], frame[:payload])
-        when :altsvc, :blocked
+        when :altsvc
+          # 4.  The ALTSVC HTTP/2 Frame
+          # An ALTSVC frame on stream 0 with empty (length 0) "Origin"
+          # information is invalid and MUST be ignored.
+          if frame[:origin] && !frame[:origin].empty?
```

Comment:
For stream you have `.nil?`check, any reason why you omitted it here?

Reply:
checking that the frame exists and is not empty was the condition, there was no gain in being explicit with `!frame[:origin].nil?`, IMO.

Reply:
So why do check for nil in stream.rb? Should we remove it there too?

Reply:
I see what you mean. I'm out til Monday, I'll sort it out by then.

Reply:
done